lib/Sema/CSSolver.cpp

```diff
@@ -1765,77 +1829,32 @@ void ConstraintSystem::partitionDisjunction(
     return true;
   });
 
-  // Now collect the overload choices that are defined within the type
-  // that was designated in the operator declaration.
-  auto designatedNominalTypes = getOperatorDesignatedNominalTypes(Choices[0]);
-  if (!designatedNominalTypes.empty()) {
-    forEachChoice(
-        Choices, [&](unsigned constraintIndex, Constraint *constraint) -> bool {
-          auto *decl = constraint->getOverloadChoice().getDecl();
-          auto *funcDecl = cast<FuncDecl>(decl);
-
-          auto *parentDecl = funcDecl->getParent()->getAsDecl();
-          for (auto designatedTypeIndex : indices(designatedNominalTypes)) {
-            auto *designatedNominal =
-                designatedNominalTypes[designatedTypeIndex];
-            if (parentDecl == designatedNominal) {
-              if (designatedTypeIndex >= definedInDesignatedType.size())
-                definedInDesignatedType.resize(designatedTypeIndex + 1);
-              auto &constraints = definedInDesignatedType[designatedTypeIndex];
-              constraints.push_back(constraintIndex);
-              return true;
-            }
+  // Local function to create the next partition based on the options
+  // passed in.
+  PartitionAppendCallback appendPartition =
+      [&](SmallVectorImpl<unsigned> &options) {
+        if (options.size()) {
+          PartitionBeginning.push_back(Ordering.size());
+          Ordering.insert(Ordering.end(), options.begin(), options.end());
+        }
+      };
 
-            if (auto *extensionDecl = dyn_cast<ExtensionDecl>(parentDecl)) {
-              parentDecl = extensionDecl->getExtendedNominal();
-              if (parentDecl == designatedNominal) {
-                if (designatedTypeIndex >=
-                    definedInExtensionOfDesignatedType.size())
-                  definedInExtensionOfDesignatedType.resize(
-                      designatedTypeIndex + 1);
-
-                auto &constraints =
-                    definedInExtensionOfDesignatedType[designatedTypeIndex];
-                constraints.push_back(constraintIndex);
-                return true;
-              }
-            }
-          }
+  SmallVector<SmallVector<unsigned, 4>, 4> definedInDesignatedType;
+  SmallVector<SmallVector<unsigned, 4>, 4> definedInExtensionOfDesignatedType;
```

Comment:
I'm not sure why would we'd want to pass these into a `partitionForDesignatedTypes` call instead of just making both local in there because they are not used `partitionDisjunction`?...

Reply:
Oversight. Will fix momentarily.